tests/test_docs/test_orm_integration/test_orm_integration.py

```diff
@@ -0,0 +1,224 @@
+    args: {}
+    class_name: Dialogues
+  strategy:
+    class_name: Strategy
+    args:
+      total_price: 10
+      seller_tx_fee: 0
+      currency_id: 'FET'
+      ledger_id: 'fetchai'
+      is_ledger_tx: True
+      has_data_source: True
+      data_for_sale: {}
+      search_schema:
+        attribute_one:
+          name: country
+          type: str
+          is_required: True
+        attribute_two:
+          name: city
+          type: str
+          is_required: True
+      search_data:
+        country: UK
+        city: Cambridge
+dependencies:
+  SQLAlchemy: {}"""
+
+buyer_strategy_replacement = """models:
+  dialogues:
+    args: {}
+    class_name: Dialogues
+  strategy:
+    class_name: Strategy
+    args:
+      max_price: 40
+      max_buyer_tx_fee: 100
+      currency_id: 'FET'
+      ledger_id: 'fetchai'
+      is_ledger_tx: True
+      search_query:
+        search_term: country
+        search_value: UK
+        constraint_type: '=='
+ledgers: ['fetchai']"""
+
+
+ORM_SELLER_STRATEGY_PATH = Path(
+    ROOT_DIR, "tests", "test_docs", "test_orm_integration", "orm_seller_strategy.py"
+)
+
+
+class TestOrmIntegrationDocs(AEAWithOefTestCase):
+    """This class contains the tests for the orm-integration.md guide."""
+
+    @skip_test_ci
+    def test_orm_integration_docs_example(self, pytestconfig):
+        """Run the weather skills sequence."""
+        self.initialize_aea()
+
+        seller_aea_name = "my_seller_aea"
+        buyer_aea_name = "my_buyer_aea"
+        self.create_agents(seller_aea_name, buyer_aea_name)
+
+        ledger_apis = {FETCHAI: {"network": "testnet"}}
+
+        # Setup seller
+        seller_aea_dir_path = Path(self.t, seller_aea_name)
+        os.chdir(seller_aea_dir_path)
+        self.add_item("connection", "fetchai/oef:0.2.0")
+        self.add_item("skill", "fetchai/generic_seller:0.1.0")
+        self.run_install()
+        force_set_config("agent.ledger_apis", ledger_apis)
+        self.set_config("agent.default_connection", "fetchai/oef:0.2.0")
+
+        # Setup Buyer
+        buyer_aea_dir_path = Path(self.t, buyer_aea_name)
+        os.chdir(buyer_aea_dir_path)
+
+        self.add_item("connection", "fetchai/oef:0.2.0")
+        self.add_item("skill", "fetchai/generic_buyer:0.1.0")
+        self.run_install()
+        force_set_config("agent.ledger_apis", ledger_apis)
+        self.set_config("agent.default_connection", "fetchai/oef:0.2.0")
+
+        # Generate and add private keys
+        self.generate_private_key()
+        self.add_private_key()
+
+        # Add some funds to the buyer
+        self.generate_wealth()
+
+        # Update the seller AEA skill configs.
+        seller_skill_config_path = Path(
+            seller_aea_dir_path,
+            "vendor",
+            "fetchai",
+            "skills",
+            "generic_seller",
+            DEFAULT_SKILL_CONFIG_FILE,
+        )
+        seller_skill_config = yaml.safe_load(seller_skill_config_path.open())
```

Comment:
You could use this: https://github.com/fetchai/agents-aea/blob/develop/aea/test_tools/generic.py#L91

Reply:
Also, is it worth exposing this as a method on the test case?